device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -329,7 +329,7 @@ public void startDeviceTwin(IotHubEventCallback deviceTwinStatusCallback, Object
             /*
             **Codes_SRS_DEVICECLIENT_25_011: [**The function shall create a new instance of class Device Twin and request all twin properties by calling getDeviceTwin**]**
              */
-            deviceTwin = new DeviceTwin(this, this.config, deviceTwinStatusCallback, deviceTwinStatusCallbackContext,
+            deviceTwin = new DeviceTwin(this.transport, this.config, deviceTwinStatusCallback, deviceTwinStatusCallbackContext,
                                         genericPropertyCallBack, genericPropertyCallBackContext);
```

Comment:
How do you plan to initialize this transport ? It is most certainly dependent on Device client. If not so you will have to recreate logic in device client to initialize transport